ert3_examples/polynomial/poly.py

```diff
@@ -32,16 +45,21 @@ def valid_coefficients(file_path):
     return arg_parser
 
 
-def _evaluate_polynomial(coefficients):
+def _evaluate_polynomial(coefficients, x_uncertainties):
     a, b, c = coefficients["a"], coefficients["b"], coefficients["c"]
-    x_range = tuple(range(10))
-    return tuple(a * x ** 2 + b * x + c for x in x_range)
+    if x_uncertainties:
+        x_range = range(len(x_uncertainties))
+        xs = map(sum, zip(tuple(x_range), x_uncertainties))
```

Comment:
As mentioned by @ManInFez earlier `zip` will consume the two iterables until one of them is exhausted. That is:

```python
>>> tuple(zip([1, 2, 3, 4, 5, 6, 7], [2, 4, 6]))
((1, 2), (2, 4), (3, 6))
```

Which is perhaps not what is intended? If not, we can complain if the length of `x_range` and `x_uncertainties` are different...

Reply:
Not completely sure what you mean. `x_range` will be the range of `x_uncertainties`if it is present, else it will be the default `range(10)`, so we will not get a situation in which `zip` takes two iterables of different lenght. Do we want to raise an error?

Reply:
Sure, but I can configure `x_uncertainties` to be of length `20` in my configuration file right? And then you will get a length mismatch... This is not very important in an example, but I think we should avoid silently continuing on unexpected input data. I suggest we raise an exception if there is a mismatch on length 💣

Reply:
Never mind! My mistake 😬